From 75de6a79143617b9dc1c02d2d73f0723a413214a Mon Sep 17 00:00:00 2001
From: Yuya Hamamachi <yuya.hamamachi.sx@renesas.com>
Date: Tue, 13 Feb 2024 18:45:04 +0900
Subject: [PATCH] arm64: dts: renesas: Add canfd nodes for S4SK

This patch is ported from spider's patch.
And, currently, all can pins are available.
But, S4SK has only CAN0 and CAN1 output.

Signed-off-by: Yuya Hamamachi <yuya.hamamachi.sx@renesas.com>
---
 arch/arm64/boot/dts/renesas/r8a779f0-s4sk.dts | 162 ++++++++++++++++++
 1 file changed, 162 insertions(+)

diff --git a/arch/arm64/boot/dts/renesas/r8a779f0-s4sk.dts b/arch/arm64/boot/dts/renesas/r8a779f0-s4sk.dts
index 7f3645257022..4e37327b3584 100644
--- a/arch/arm64/boot/dts/renesas/r8a779f0-s4sk.dts
+++ b/arch/arm64/boot/dts/renesas/r8a779f0-s4sk.dts
@@ -51,6 +51,87 @@ vcc_sdhi: regulator-vcc-sdhi {
 	};
 };
 
+
+&canfd0 {
+	pinctrl-0 = <&canfd0_pins>, <&canfd1_pins>, <&canfd2_pins>,
+				<&canfd3_pins>, <&canfd4_pins>, <&canfd5_pins>,
+				<&canfd6_pins>, <&canfd7_pins>;
+	pinctrl-names = "default";
+	status = "okay";
+
+	channel0 {
+		status = "okay";
+	};
+
+	channel1 {
+		status = "okay";
+	};
+
+	channel2 {
+		status = "okay";
+	};
+
+	channel3 {
+		status = "okay";
+	};
+
+	channel4 {
+		status = "okay";
+	};
+
+	channel5 {
+		status = "okay";
+	};
+
+	channel6 {
+		status = "okay";
+	};
+
+	channel7 {
+		status = "okay";
+	};
+};
+
+&canfd1 {
+	pinctrl-0 = <&canfd8_pins>,<&canfd9_pins>, <&canfd10_pins>, 
+				<&canfd11_pins>, <&canfd12_pins>, <&canfd13_pins>, 
+				<&canfd14_pins>, <&canfd15_pins>;
+	pinctrl-names = "default";
+	status = "okay";
+
+	channel0 {
+		status = "okay";
+	};
+
+	channel1 {
+		status = "okay";
+	};
+
+	channel2 {
+		status = "okay";
+	};
+
+	channel3 {
+		status = "okay";
+	};
+
+	channel4 {
+		status = "okay";
+	};
+
+	channel5 {
+		status = "okay";
+	};
+
+	channel6 {
+		status = "okay";
+	};
+
+	channel7 {
+		status = "okay";
+	};
+};
+
 &extal_clk {
 	clock-frequency = <20000000>;
 };
@@ -113,6 +194,87 @@ &pfc {
 	pinctrl-0 = <&scif_clk_pins>;
 	pinctrl-names = "default";
 
+	canfd0_pins: canfd0 {
+		groups = "canfd0_data";
+		function = "canfd0";
+	};
+
+	canfd1_pins: canfd1 {
+		groups = "canfd1_data";
+		function = "canfd1";
+	};
+
+	canfd2_pins: canfd2 {
+		groups = "canfd2_data";
+		function = "canfd2";
+	};
+
+	canfd3_pins: canfd3 {
+		groups = "canfd3_data";
+		function = "canfd3";
+	};
+
+	canfd4_pins: canfd4 {
+		groups = "canfd4_data";
+		function = "canfd4";
+	};
+
+	canfd5_pins: canfd5 {
+		groups = "canfd5_data";
+		function = "canfd5";
+	};
+
+	canfd6_pins: canfd6 {
+		groups = "canfd6_data";
+		function = "canfd6";
+	};
+
+	canfd7_pins: canfd7 {
+		groups = "canfd7_data";
+		function = "canfd7";
+	};
+
+	canfd8_pins: canfd8 {
+		groups = "canfd8_data";
+		function = "canfd8";
+	};
+
+	canfd9_pins: canfd9 {
+		groups = "canfd9_data";
+		function = "canfd9";
+	};
+
+	canfd10_pins: canfd10 {
+		groups = "canfd10_data";
+		function = "canfd10";
+	};
+
+	canfd11_pins: canfd11 {
+		groups = "canfd11_data";
+		function = "canfd11";
+	};
+
+	canfd12_pins: canfd12 {
+		groups = "canfd12_data";
+		function = "canfd12";
+	};
+
+	canfd13_pins: canfd13 {
+		groups = "canfd13_data";
+		function = "canfd13";
+	};
+
+	canfd14_pins: canfd14 {
+		groups = "canfd14_data";
+		function = "canfd14";
+	};
+
+	canfd15_pins: canfd15 {
+		groups = "canfd15_data";
+		function = "canfd15";
+	};
+
+
 	i2c2_pins: i2c2 {
 		groups = "i2c2";
 		function = "i2c2";
-- 
2.34.1

